base/special/log.jl

```diff
@@ -397,51 +397,167 @@ function log1p(x::Float32)
     end
 end
 
-
-@inline function log_ext_kernel(x_hi::Float64, x_lo::Float64)
-    c1hi = 0.666666666666666629659233
-    hi_order =  evalpoly(x_hi, (0.400000000000000077715612, 0.285714285714249172087875,
-                                0.222222222230083560345903, 0.181818180850050775676507,
-                                0.153846227114512262845736, 0.13332981086846273921509,
-                                0.117754809412463995466069, 0.103239680901072952701192,
-                                0.116255524079935043668677))
-    res_hi, res_lo = two_mul(hi_order, x_hi)
-    res_lo = fma(x_lo, hi_order, res_lo)
-    ans_hi = c1hi + res_hi
-    ans_lo = ((c1hi - ans_hi) + res_hi) + (res_lo + 3.80554962542412056336616e-17)
-    return ans_hi, ans_lo
-end
+const t_log_ext_Float64 = (
```

Comment:
LUTs are tricky to benchmark. They do well on microbenchmarks, but the additional cache pressure can cause problems. They also don't vectorize well. https://stackoverflow.com/questions/41529921/using-simd-on-amd64-when-is-it-better-to-use-more-instructions-vs-loading-from has some discussion.

Reply:
pow has way too many branches to vectorize well anyway. I agree that using a 64 element table is probably worthwhile since it would shrink the size of the LUT from 3Kb to 1.5Kb. That said, we already use a LUT for `exp`, and @chriselrod has shown that table based implementations can vectorize very well. Getting a good set of benchmarks on this is important, but I do pretty firmly believe that this is a better implementation that the current one.

Reply:
Sure, but my suggestion would be to keep the LUT-less version around somewhere, so it's easy to benchmark in end-to-end applications to see the effect of the cache pressure.

Reply:
This approach is also fast enough that I'm going to try using the same implementation for regular `log` (with some changes to lower the precision) which would save a table of the same size.

Reply:
I've done some initial testing that confirms that this method should be competitive for regular `log` functions.

Reply:
> based implementations can vectorize very well.

That of course requires gather instructions, which requires at least AVX2 on x86 and SVE on Aarch64.
So basically any recent x86 CPU with FMA can also gather, and while SVE2 is standard in ARM9, this means LUTs wont vectorize on ARM8 (except for the A64FX).

That's all mute here if your pow implementation isn't vectorizing anyway.
But it would be great to have a vectorizable implementation, e.g. maybe the `@fastmath` variant?

Also, I just checked, and `@inline exp(x)` no longer vectorizes on Julia master. It did briefly after callsite inlining was introduced.

Reply:
that's unfortunate.